modules/sr/robot/utils.py

```diff
@@ -7,3 +21,30 @@ def map_to_range(
 ) -> float:
     """Maps a value from within one range of inputs to within a range of outputs."""
     return ((value - old_min) / (old_max - old_min)) * (new_max - new_min) + new_min
+
+
+TDevice = TypeVar('TDevice', bound=Device)
+
+
+def get_robot_device(robot: Robot, name: str, kind: Type[TDevice]) -> TDevice:
+    device: Optional[Device] = None
+    try:  # webots 2020b fails to assign the output to the expected type raising TypeError
```

Comment:
I'm not sure this comment is quite correct. Perhaps let's just describe this as:
```suggestion
    try:
        # webots 2020b is buggy and always TypeError, however we're aiming to
        # be forwards compatible with 2021a, so try this first.
```